src/index.js

```diff
@@ -1,25 +1,80 @@
+const loaderUtils = require("loader-utils");
 const htmlparser = require("htmlparser");
+const defaultOptions = {
+  cssModules: true,
+  mangleIds: true,
+  idPattern: 'svg-[name]-[sha512:hash:base64:7]',
+};
+
 
-function assembleNode(node) {
+const getOptions = context => Object.assign({},
+  loaderUtils.getOptions(context),
+  defaultOptions
+);
+
+function getIdReplacement(context, options, id) {
+  return loaderUtils.interpolateName(context, options.idPattern, { content: id }).replace(/[^a-zA-Z0-9]/g,'-');
+}
+
+function assembleNode(context, options, node, root) {
     if (node.type === 'text') {
-        return `'${node.data.replace(/\n/g, '\\n')}'`;
+        return JSON.stringify(node.data)
```

Comment:
this was a bug, any svg using `'` in a text node would crash on import, and `\` would get hidden because it was interpreted